plaid-stl/src/datetime.rs

```diff
@@ -1,18 +1,64 @@
-use chrono::{DateTime, Utc};
+use chrono::{NaiveDate, Utc, DateTime};
 use serde::{Deserialize, Deserializer};
 
-// Custom deserializer for an optional DateTime<Utc>
-pub fn deserialize_option_timestamp<'de, D>(deserializer: D) -> Result<Option<DateTime<Utc>>, D::Error>
+// Custom deserializer from a ISO 8601 string to a DateTime<Utc>
+pub fn deserialize_rfc3339_timestamp<'de, D>(deserializer: D) -> Result<DateTime<Utc>, D::Error>
 where
     D: Deserializer<'de>,
 {
-    // Try to deserialize as a string (ISO 8601 format) or return None if missing
-    let opt = Option::<String>::deserialize(deserializer)?;
+    let s = String::deserialize(deserializer)?;
 
-    // Parse the timestamp string into a DateTime<Utc>, if it's present
-    Ok(opt.and_then(|s| {
+    Ok(
         DateTime::parse_from_rfc3339(&s)
-            .ok()
-            .map(|dt| dt.with_timezone(&Utc))
-    }))
+            .map_err(serde::de::Error::custom)?
+            .with_timezone(&Utc)
+    )
 }
+
+// Custom deserializer from a possibly empty ISO 8601 string to an optional DateTime<Utc>
+pub fn deserialize_option_rfc3339_timestamp<'de, D>(deserializer: D) -> Result<Option<DateTime<Utc>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    // If the value is empty, we return None
+    let s = match Option::<String>::deserialize(deserializer)? {
+        Some(v) => v,
+        None => return Ok(None),
+    };
+
+    let s = s.trim();
+    // If the value is an empty string, we assume the default value has been returned instead of
+    // throwing parsing errors
+    if s.is_empty() {
+        return Ok(None)
+    }
+
+    Ok(Some(
+        DateTime::parse_from_rfc3339(&s)
+            .map_err(serde::de::Error::custom)?
+            .with_timezone(&Utc)
+    ))
```

Comment:
@michelemin  If the string is empty, I return None. Before, if the string is not empty and parsing fails, the deserializer returns None, which I think is misleading. Now, if parsing fails on an non-empty string, the deserializer will throw an error.